lib/pure/httpclient.nim

```diff
@@ -1210,7 +1210,8 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   for i in 1..client.maxRedirects:
     if result.status.redirection():
       let redirectTo = getNewLocation(lastURL, result.headers)
-      result = await client.requestAux(redirectTo, httpMethod, body, headers)
+      var meth = if result.status != "307": "GET" else: httpMethod
```

Comment:
This requires a comment. Why does status 307 enforce a "GET"?

Reply:
It's actually the opposite. See: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_Redirection

I reversed the order of the check to be more clear, but I'm just repeating the behaviour of the previous implementation of POST